src/components/Pega_Extensions_SecureRichText/index.tsx

```diff
@@ -0,0 +1,229 @@
+  const {
+    getPConnect,
+    value = '',
+    placeholder = '',
+    validatemessage = '',
+    label,
+    hideLabel = false,
+    helperText = '',
+    testId,
+    displayMode,
+    additionalProps = {},
+    isTableFormatter = false,
+    fieldMetadata
+  } = props;
+
+  const { formatter } = props;
+  const pConn = getPConnect();
+  const editorRef = useRef<EditorExtensionState>(null);
+  const fieldAdditionalInfo = fieldMetadata?.additionalInformation;
+  const additionalInfo = fieldAdditionalInfo
+    ? {
+        content: fieldAdditionalInfo
+      }
+    : undefined;
+  const toasterContext = useToaster();
+  const actionSequencer = useMemo(() => PCore.getActionsSequencer(), []);
+  const form: Features['form'] & { enableRTEImageAttachments?: boolean } =
+    PCore.getEnvironmentInfo().environmentInfoObject?.features?.form || {
+      attachmentPageInstructionEnabled: false
+    };
+  const { enableRTEImageAttachments = false } = form;
+
+  let { readOnly, required, disabled } = props;
+  [readOnly, required, disabled] = [readOnly, required, disabled].map(
+    prop => prop === true || (typeof prop === 'string' && prop === 'true')
+  );
+
+  useEffect(() => {
+    editorRef.current?.insertHtml(value, true);
+  }, [value]);
+
+  const updatedRteContent = useMemo(() => {
+    return updateContentWithAbsoluteURLsOfImgSrcs(value, pConn);
+  }, [value, pConn]);
+
+  const displayComponent = updatedRteContent ? (
+    <RichTextViewer content={updatedRteContent} type='html' />
+  ) : (
+    <NoValue />
+  );
+
+  if (displayMode === 'DISPLAY_ONLY' && formatter) {
+    if (isTableFormatter && formatExists(formatter)) {
+      return textFormatter(formatter, value);
+    }
+
+    return displayComponent;
+  }
+
+  let richTextComponent;
+
+  if (readOnly) {
+    // Rich Text read-only component
+    richTextComponent = hideLabel ? (
+      displayComponent
+    ) : (
+      <Details>
+        <FieldValueItem name={label} value={displayComponent} variant='stacked' />
+      </Details>
+    );
+  } else {
+    // Rich Text editable component
+    const actions = pConn.getActions();
+    const actionsApi = pConn.getActionsApi();
+    let status = '';
+    if (validatemessage !== '') {
+      status = 'error';
+    }
+    const handleChange = () => {
+      if (status === 'error') {
+        const property = pConn.getStateProps().value;
+        pConn.clearErrorMessages({
+          property
+        });
+      }
+    };
+
+    const handleBlur = () => {
+      if (editorRef.current) {
+        const editorValue = editorRef.current.getHtml();
+        const property = pConn.getStateProps().value;
+        if (!editorValue) {
+          pConn.getValidationApi().validate(editorValue);
+        }
+        if (value !== editorValue) {
+          handleEvent(actionsApi, 'changeNblur', property, editorValue);
+        }
+      }
+    };
+
+    const onImageAdded: EditorProps['onImageAdded'] = (image, id) => {
```

Comment:
@Akshansr why do we need this function if you cannot add images?

Reply:
Removed all image handlers @ricmars

Reply:
We added sanitisation to filter out all images and links from content before its sent to server